HeterogeneousCore/SonicTriton/test/tritonTest_cfg.py

```diff
@@ -29,6 +29,40 @@
   "TritonGraphProducer": "gat_test",
 }
 
+inConvs = {
+  "TritonImageProducer": cms.VPSet(
```

Comment:
To simplify this for users, we might want to add some additional infrastructure:
1. To handle default cases, we could map the standard converters to the appropriate Triton data type names/enums (see the TritonData constructor). Then these wouldn't need to be specified in the python config explicitly.
2. We might want to investigate if there's a way to augment the model config and protocols so that, when a FaaST server is used, it could automatically specify (probably via some gRPC message) when an alternate converter needs to be used (and which one).

Item 2 assumes that there would not be a case where the client wants to use a different converter than specified by the model in the repository/on the server.

Reply:
One idea to avoid a) overloading the meaning of some existing protobuf field (potentially dangerous) or b) extending the proto definitions (maintenance burden):
we could come up with a policy whereby the input and output names in the config.pbtxt are extended to indicate which non-standard converter (if any) is needed. Then these extended names could be handled by the client.
We could try to have the FaaST server perform this extension automatically (in order to keep the same config.pbtxt files for GPU and FPGA versions of a model). But since other model files might need to be different for FPGA usage, changing the config.pbtxt might be okay.

Even with this, it could still be useful to have the option introduced here to specify a converter manually (for testing).

Reply:
1) Ok yeah, this is a good suggestion, Ill set this up.

2) Hmm, perhaps we could use something existing like the `platform` field in `ModelMetadataResponse`, and fill it with the server's desired converter. This seems simple enough for FaaST, but I'm not sure how easy this would be with a standard GPU Triton server.

Reply:
There's still only one `platform` per model, so we would have to pack all the information for every input and output into that one string, which isn't ideal.

Reply:
Oh yeah, I keep forgetting that other models have multiple inputs/outputs.

Reply:
So this now uses defaults successfully when they are available, so 1) is addressed.

For 2), Ok, I missed your suggestion of the model name field. I think the version field might be even more accurate but it's a minor point. But can we enforce a policy whereby we use the field to encode the necessary converter? I'm wondering if you think this is actually feasible or if you were only thinking out loud a bit here.

Reply:
To be a bit more explicit, the most automatic implementation would be:
1. config.pbtxt retains the normal input/output names (same for GPU and FPGA)
2. FaaST appends information to the input/output names when returning the `ModelMetadataResponse`, e.g. "input1_FPGAConverter:foo"
3. TritonClient knows to check for the presence of "_FPGAConverter:foo"

The question here would be how "foo" is provided for each model input/output. Maybe there could be a separate section of the config.pbtxt; we have to learn more about how Triton parses it and what (if anything) it ignores.

The alternative would be a bit more manual on the user side:
1. make a separate FPGA copy of the GPU config.pbtxt, and add "_FPGAConverter:foo" to the input/output names.
2. FaaST uses the FPGA copy of the config.pbtxt (maybe this could also be automated...)